medicaltorch/datasets.py

```diff
@@ -199,41 +231,142 @@ class MRI2DSegmentationDataset(Dataset):
     :param cache: if the data should be cached in memory or not.
     :param transform: transformations to apply.
     """
-    def __init__(self, filename_pairs, slice_axis=2, cache=True,
-                 transform=None, slice_filter_fn=None, canonical=False):
-        self.filename_pairs = filename_pairs
+    def __init__(self, root_dir, subject_lst, target_suffix, contrast_lst, contrast_balance={},
```

Comment:
Mmm that's tricky because... I am not sure that it will be of interest for perone/medicaltorch... And we do not want to move from ivadomed to medicaltorch-neuropoly if at the end we have to move back things from medicaltorch-neuropoly to ivadomed because not relevant to be merged into perone/medicaltorch.
Not sure what is the best way to go..

For this specific class --> I would suggest to keep our implementation in ivadomed because very specific: BIDS, FlipAngle etc., contrast balance, ROI...etc.

Reply:
Maybe we could keep the constructor (very specific) of this class in `ivadomed` while modifying the other methods here in `medicaltorch-neuropoly` (since the modifications, ROI & multichannel) are more likely to be integrated into `perone/medicaltorch`.